pairtools/pairtools_dedup.py

```diff
@@ -1,248 +1,421 @@
+        )
+
     if not output_unmapped:
         outstream_unmapped = None
-    elif (output_unmapped == '-' or 
-        (pathlib.Path(output_unmapped).absolute() == pathlib.Path(output).absolute())):
+    elif output_unmapped == "-" or (
+        pathlib.Path(output_unmapped).absolute() == pathlib.Path(output).absolute()
+    ):
         outstream_unmapped = outstream
-    elif (pathlib.Path(output_unmapped).absolute() == pathlib.Path(output_dups).absolute()):
+    elif (
+        pathlib.Path(output_unmapped).absolute() == pathlib.Path(output_dups).absolute()
+    ):
         outstream_unmapped = outstream_dups
     else:
-        outstream_unmapped = _fileio.auto_open(output_unmapped, mode='w', 
-                                            nproc=kwargs.get('nproc_out'),
-                                            command=kwargs.get('cmd_out', None))
-                             
+        outstream_unmapped = _fileio.auto_open(
+            output_unmapped,
+            mode="w",
+            nproc=kwargs.get("nproc_out"),
+            command=kwargs.get("cmd_out", None),
+        )
 
     header, body_stream = _headerops.get_header(instream)
     header = _headerops.append_new_pg(header, ID=UTIL_NAME, PN=UTIL_NAME)
-
     if send_header_to_dedup:
-        outstream.writelines((l+'\n' for l in header))
+        outstream.writelines((l + "\n" for l in header))
     if send_header_to_dup and outstream_dups and (outstream_dups != outstream):
-        outstream_dups.writelines((l+'\n' for l in header))
-    if (outstream_unmapped and (outstream_unmapped != outstream) 
-            and (outstream_unmapped != outstream_dups)):
-        outstream_unmapped.writelines((l+'\n' for l in header))
+        dups_header = header
+        dups_header[-1] += " parent_readID"
+        outstream_dups.writelines((l + "\n" for l in dups_header))
+    if (
+        outstream_unmapped
+        and (outstream_unmapped != outstream)
+        and (outstream_unmapped != outstream_dups)
+    ):
+        outstream_unmapped.writelines((l + "\n" for l in header))
 
     column_names = _headerops.extract_column_names(header)
     extra_cols1 = []
     extra_cols2 = []
     if extra_col_pair is not None:
         for col1, col2 in extra_col_pair:
             extra_cols1.append(
-                int(col1) if col1.isdigit() else column_names.index(col1))
+                int(col1) if col1.isdigit() else column_names.index(col1)
+            )
             extra_cols2.append(
-                int(col2) if col2.isdigit() else column_names.index(col2))
-                            
-
-    streaming_dedup( method, max_mismatch, sep,
-        c1, c2, p1, p2, s1, s2, extra_cols1, extra_cols2, unmapped_chrom,
-        body_stream, outstream, outstream_dups,
-        outstream_unmapped, out_stat, mark_dups)
+                int(col2) if col2.isdigit() else column_names.index(col2)
+            )
+
+    if backend == "cython":
+        streaming_dedup(
```

Comment:
Should we rename it to streaming_dedup_cython?

Reply:
Sounds good!